Update pickle.rst (#24776)
python/cpython: Doc/library/pickle.rst

The following types can be pickled:

* ``None``, ``True``, and ``False``;

* integers, floating-point numbers, complex numbers;

* strings, bytes, bytearrays;

* tuples, lists, sets, and dictionaries containing only picklable objects;

* functions (built-in and user-defined) defined at the top level of a module
  (using :keyword:`def`, not :keyword:`lambda`);

> * classes defined at the top level of a module;

Reviewer: This isn't actually true, classes nested in other classes work fine. Opened https://bugs.python.org/issue47206 to fix this.

Author: Good catch! Should I fix it in this PR?

Reviewer: Let's do it separately, so we can keep this to non-substantive changes

   f = io.BytesIO()
   p = MyPickler(f)

does the same but all instances of ``MyPickler`` will by default
share the private dispatch table.  On the other hand, the code ::

   copyreg.pickle(SomeClass, reduce_SomeClass)
   f = io.BytesIO()
   p = pickle.Pickler(f)

> does the same but all instances of :class:`pickle.Pickler` will by default
> share the modified global dispatch table.

Reviewer: Is this change intentional? Doesn't look like a typo fix, though I agree that the old wording doesn't make much sense. However, I would word it as 
```suggestion
modifies the global dispatch table shared by all users of the :mod:`copyreg` module.
```

Author: Yes this was intentional but your wording is better, thanks!
